app/models/ingestible.rb

```diff
@@ -25,6 +25,12 @@ class Ingestible < ApplicationRecord
 
   # after_commit :update_parsing # this results in ActiveStorage::FileNotFoundError in dev/local storage
 
+  before_save do
+    if @texts.present?
```

Comment:
if we modified parsed texts array we serialize it back to JSON before saving ingestible